src/TRestGeant4AnalysisProcess.cxx

```diff
@@ -387,6 +387,11 @@ void TRestGeant4AnalysisProcess::InitProcess() {
             fTracksEDepObservables.push_back(fObservables[i]);
             fParticleTrackEdep.emplace_back(particleName.Data());
         }
+
+        if (fObservables[i].find("primaryOriginDistanceToPrism") != string::npos) {
```

Comment:
This is a metadata member parameter, so it is not really needed to explicitly initialise inside the code, since it will be automatically done by `TRestEventProcessLLInitFromnConfigFile` I believe.

Reply:
I tried but it does not initialize so I am keeping this.

Reply:
But then why not do that at `InitFromConfigFile` which is the natural place?

Reply:
This should work out of the box, since `InitFromConfigFile` has not been re-implemented in this class.

See for example this process, it defines a `TVector3` and it is properly initialised without any additional line of code.

https://github.com/rest-for-physics/axionlib/blob/master/inc/TRestAxionEventProcess.h

You should initialize these variables to `(0,0,0)` and just let the used define them in any case, even in the case the observable was not defined.